File: 506-Relative Ranks-相对名次/506-Relative Ranks-相对名次.js
/**
 * @param {number[]} score
 * @return {string[]}
 */
 var findRelativeRanks = function(score) {
    const res = []
    let len = score.length
    // let preScore = JSON.parse(JSON.stringify((score)))
    const preScore = []
    for(let i = 0; i < len; i++){
        preScore[i] = score[i]
    }
    score.sort((a, b) => b-a)
    for(let i = 0; i < len; i++){
        let sortIndex = score.indexOf(preScore[i])
        if(sortIndex  === 0){
            res[i] = "Gold Medal"
        }else if(sortIndex === 1){
            res[i] = "Silver Medal"
        }else if(sortIndex === 2){
            res[i] = "Bronze Medal"
        }else{
            res[i] = String(sortIndex + 1)
        }
    }
    return res;
};

File: 506-Relative Ranks-相对名次/506-Relative Ranks-相对名次.java
class Solution {
    public String[] findRelativeRanks(int[] score) {
        int len = score.length;
    	int[][] res = new int[len][2];
    	String[] desc = {"Gold Medal", "Silver Medal", "Bronze Medal"};
    	for(int i = 0; i < len; i++) {
    		res[i][0] = score[i];
    		res[i][1] = i;
    	}	
    	Arrays.sort(res, (a, b) -> b[0] - a[0]);
    	String[] ans = new String[len];
    	for(int i = 0; i < len; i++) {
    		if(i >= 3) {
    			ans[res[i][1]] = Integer.toString(i + 1);
    		}else {
    			ans[res[i][1]] = desc[i];
    		}
    	}
		return ans;
    }
}
